### ISSUE TITLE
Crash while crafting advanced solar generator

### ISSUE THREAD

**Issue by MatthiasMann** [aidancbrady/Mekanism#5004]
#### Issue description:
Tryign to craft the advanced solar generator in a tinkers crafting table causes the game to freeze for several seconds and then crash with a rendering stack trace of a random block in the world.

#### Steps to reproduce:
1. Use a tinkers crafting table
2. Put the items for the advanced generator in the table (or use JEI '+' click with all items in the inventory)
3. Game freeze => crash

The same crash happens again when restarting and opening the crafting table GUI - even if all items are hidden in JEI by entering an non existing string in the search bar before going into the crafting table.

#### Version (make sure you are on the latest version before reporting):
1.12.2-forge1.12.2-14.23.3.2668
Mekanism-1.12.2-9.4.3.330.jar
MekanismGenerators-1.12.2-9.4.3.330.jar
MekanismTools-1.12.2-9.4.3.330.jar
TConstruct-1.12.2-2.9.1.65.jar

#### If a (crash)log is relevant for this issue, link it here: (_It's almost always relevant_)
[crash-2018-04-18_23.46.41-client.txt](https://github.com/aidancbrady/Mekanism/files/1925961/crash-2018-04-18_23.46.41-client.txt)
[crash-2018-04-19_00.13.08-client.txt](https://github.com/aidancbrady/Mekanism/files/1925962/crash-2018-04-19_00.13.08-client.txt)

**Comment by thiakil** [comment]
Neither of those appear to be Mekanism

1st is something Silent Gems,
2nd is a Tinker's tank rendering thing

**Comment by MatthiasMann** [comment]
This is caused by having the advanced solar generator in the tinker table being rendered - when I used a normal crafting table to craft it no crash appeared. But as other items (including others from Mekanism) can be crafted without issues in the tinker table this seems to be some state corruption in the rendering system of minecraft from this advanced solar generator.

The first crash happened when I first clicked the ASG into the table using JEI '+'.
Then I removed all chaos pylons from my world and opened the tinker crafting table GUI again - that's the 2nd crash.

So this is clearly caused by the combination of the ASG being crafted in the tinker table.
